### PR TITLE
ovl_Bg_Zg OK

### PR DESCRIPTION
**Pull request by shawlucas**

### CODE REVIEW

**Comment by shawlucas** [comment]
Okay I’ll do that and update my PR shortly

**Comment by shawlucas** [comment]
Ran clang formatting on files

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
+static void func_808C0CD4(BgZg* this, GlobalContext* globalCtx);
+static void func_808C0D08(BgZg* this, GlobalContext* globalCtx);
+static void func_808C0EEC(BgZg* this, GlobalContext* globalCtx);
 
-/*
-const ActorInit Bg_Zg_InitVars =
-{
+static const ActorFunc funcTbl[] = { (ActorFunc)func_808C0CD4, (ActorFunc)func_808C0D08 };
+
+static InitChainEntry initChain[] = { ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP) };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: Always have trailing commas in init chain arrays so clang-format handles it properly. And I would have trailing commas in the function tables as well.

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 2]
```diff
@@ -22,23 +38,108 @@ const ActorInit Bg_Zg_InitVars =
...
+}
+
+static void func_808C0C50(BgZg* this) {
+    Audio_PlaySoundGeneral(NA_SE_EV_METALDOOR_OPEN, &this->dyna.actor.unk_E4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+}
+
+static s32 func_808C0C98(BgZg* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->dyna.actor;
+    s32 flag;
+    flag = (thisx->params >> 8) & 0xFF;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: This is really minor but I would have the flag assignment on the same line if you're doing it for thisx.

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.h [hunk 3]
```diff
@@ -3,10 +3,12 @@
...
 #include <ultra64.h>
 #include <global.h>
+#include <vt.h>
 
 typedef struct {
-    /* 0x0000 */ Actor actor;
-    /* 0x014C */ char unk_14C[0x20];
+    /* 0x0000 */ DynaPolyActor dyna;
+    /* 0x0164 */ s32 funcIndex;
+    /* 0x0168 */ s32 unk_168;
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.h]: 0x164 would be `action` and 0x168 would be `drawConfig`, with local variables updated in the different functions where they are used.

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 4]
```diff
@@ -22,23 +38,108 @@ const ActorInit Bg_Zg_InitVars =
...
+    DynaPolyInfo_Alloc(&D_060011D4, &local_c);
+    this->dyna.dynaPolyId = DynaPolyInfo_RegisterActor(globalCtx, &globalCtx->colCtx.dyna, thisx, local_c);
+    if ((func_808C0CC8(this) == 8) || (func_808C0CC8(this) == 9)) {
+        thisx->scale.x = thisx->scale.x * 1.3f;
+        thisx->scale.z = thisx->scale.z * 1.3f;
+        thisx->scale.y = thisx->scale.y * 1.2f;
+    }
+
+    else {
+    }
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: I don't think this is necessary?

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 5]
```diff
@@ -1,16 +1,32 @@
...
+static s32 func_808C0C98(BgZg* this, GlobalContext* globalCtx);
+static s32 func_808C0CC8(BgZg* this);
+static void func_808C0CD4(BgZg* this, GlobalContext* globalCtx);
+static void func_808C0D08(BgZg* this, GlobalContext* globalCtx);
+static void func_808C0EEC(BgZg* this, GlobalContext* globalCtx);
 
-/*
-const ActorInit Bg_Zg_InitVars =
-{
+static const ActorFunc funcTbl[] = { (ActorFunc)func_808C0CD4, (ActorFunc)func_808C0D08 };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: This array would be `actionFuncs`, indexed by the `action` variable/member.

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 6]
```diff
@@ -1,16 +1,32 @@
...
+
+static InitChainEntry initChain[] = { ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP) };
+
+static const ActorFunc funcTbl2[] = { (ActorFunc)func_808C0EEC };
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]: And this one would be `drawFuncs` or something like that.

File: src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c [hunk 7]
```diff
@@ -22,23 +38,108 @@ const ActorInit Bg_Zg_InitVars =
...
+    gSPDisplayList(gfxCtx->polyOpa.p++, &D_06001080);
+    func_800C6B54(gfxArr, gfxCtx, "../z_bg_zg.c", 320);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Zg/func_808C0EEC.s")
+static void BgZg_Draw(BgZg* this, GlobalContext* globalCtx) {
+    s32 funcIndex = this->unk_168;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Zg/BgZg_Draw.s")
+    if (((funcIndex < 0) || (0 < funcIndex)) || funcTbl2[funcIndex] == 0) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_Bg_Zg/z_bg_zg.c]:
```suggestion
    if (((funcIndex < 0) || (funcIndex > 0)) || funcTbl2[funcIndex] == NULL) {
```

**Comment by shawlucas** [comment]
fixed

**Pull request merged**